packages/graph-retriever/src/graph_retriever/strategies/base.py

```diff
@@ -13,23 +13,36 @@
 class NodeTracker:
     """Helper class for tracking traversal progress."""
 
-    def __init__(self) -> None:
-        self.visited_ids: set[str] = set()
+    def __init__(self, select_k: int, max_depth: int| None) -> None:
+        self._select_k: int = select_k
+        self._max_depth: int | None = max_depth
+        self._visited_nodes: set[int] = set()
         self.to_traverse: dict[str, Node] = {}
         self.selected: dict[str, Node] = {}
 
+    @property
+    def remaining(self):
+        """The remaining number of nodes to be selected"""
+        return max(self._select_k - len(self.selected), 0)
+
     def select(self, nodes: dict[str, Node]) -> None:
         """Select nodes to be included in the result set."""
         self.selected.update(nodes)
```

Comment:
Should `select` return the number of *new* nodes (or the number of total nodes)?

I guess most use cases would be better off with `len(tracker.selected)`. Although, we could have a `@property def num_selected(self) -> int`?